pkg/storaged/resize.jsx

```diff
@@ -0,0 +1,441 @@
+                                                                 to_fit ? grow_size : vals.size,
+                                                                 info.grow_needs_unmount,
+                                                                 vals.passphrase || recovered_passphrase)
+                                    .catch(request_passphrase_on_error_handler(dlg, vals, recovered_passphrase, block)));
+                        });
+            }
+        },
+        Inits: [
+            init_active_usage_processes(client, usage),
+            passphrase_fields.length
+                ? init_existing_passphrase(block, false, pp => { recovered_passphrase = pp })
+                : null
+        ]
+    });
+}
+
+export function shrink_dialog(client, lvol_or_part, info, to_fit) {
+    let title, block, name, orig_size, round_size;
+
+    if (lvol_or_part.iface == "org.freedesktop.UDisks2.LogicalVolume") {
+        const vgroup = client.vgroups[lvol_or_part.VolumeGroup];
+
+        title = _("Shrink logical volume");
+        block = client.lvols_block[lvol_or_part.path];
+        name = lvol_or_part.Name;
+        orig_size = lvol_or_part.Size;
+        round_size = vgroup.ExtentSize;
+    } else {
+        title = _("Shrink partition");
+        block = client.blocks[lvol_or_part.path];
+        name = block_name(block);
+        orig_size = lvol_or_part.Size;
+        round_size = 1024 * 1024;
+    }
+
+    const usage = get_active_usage(client, block && !to_fit && info.shrink_needs_unmount ? block.path : null,
+                                   _("shrink"));
+
+    if (usage.Blocking) {
+        dialog_open({
+            Title: cockpit.format(_("$0 is in use"), name),
+            Body: BlockingMessage(usage)
+        });
+        return;
+    }
+
+    let shrink_size;
+    let size_fields = [];
+    if (!to_fit) {
+        size_fields = [
+            SizeSlider("size", _("Size"),
+                       {
+                           value: orig_size,
+                           max: orig_size,
+                           round: round_size,
+                       })
+        ];
+    } else {
+        const crypto = client.blocks_crypto[block.path];
+        const cleartext = client.blocks_cleartext[block.path];
+        let content_path = null;
+        let crypto_overhead = 0;
+
+        if (crypto) {
+            if (crypto.MetadataSize !== undefined && cleartext) {
+                content_path = cleartext.path;
+                crypto_overhead = crypto.MetadataSize;
+            }
+        } else {
+            content_path = block.path;
+        }
+
+        const fsys = client.blocks_fsys[content_path];
+        if (fsys)
+            shrink_size = fsys.Size + crypto_overhead;
+
+        const vdo = client.legacy_vdo_overlay.find_by_backing_block(client.blocks[content_path]);
+        if (vdo)
+            shrink_size = vdo.physical_size + crypto_overhead;
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-19385-20230926-133853-59d8b5ea-fedora-38-devel//Coverage/lcov/github-pr.diff.gcov.html)